src/server/accounting.c

```diff
@@ -1197,76 +966,46 @@ account_resvstart(resc_resv *presv)
 void
 account_jobend(job *pjob, char *used, int type)
 {
-	int i = 0;
-	int len = 0;
-	char *pb = NULL;
-	struct svrattrl *patlist = NULL;
-	char *resc_used;
-	int resc_used_size = 0;
-	pbs_list_head temp_head;
+	char		*str = NULL;
+	char		errtime[] = "00:00:00";
+	struct		svrattrl *patlist = NULL;
+	char		*resc_used;
+	int		resc_used_size = 0;
+	pbs_list_head	temp_head;
+#ifdef WIN32
+	char		*win_str = NULL;
+#endif
+	char		tempstr[40] = {0};
 
 	CLEAR_HEAD(temp_head);
 	/* pack in general information about the job */
 
-	pb = acct_job(pjob, type, acct_buf, acct_bufsize);
-	len = acct_bufsize - (pb - acct_buf);
-
-	/*
-	 * for each keyword=value pair added, the following steps should be
-	 * followed:
-	 * a. calculate (or over-estimate) the size of the date to be added
-	 * b. check that there is sufficient room in the buffer, "len" is the
-	 *    current unused amount.
-	 * c. If necessary, grow the buffer by calling grow_acct_buf().
-	 *    If this function fails,  just write out what we already have.
-	 * d. Append the new datum to the buffer at "pb".  Each new item should
-	 *    have a single leading space.  The variable "pb" is maintained
-	 *    to point to the end to save "strcat" time.
-	 * e. Advance "pb" by the length of the datum added and decrement
-	 *    "len" by the same amount.
-	 */
+	acct_buf[0] = '\0';
+	acct_job(pjob, type, &acct_buf, &acct_bufsize);
 
 	/* session */
-	i = 30;
-	if (i > len)
-		if (grow_acct_buf(&pb, &len, i) == -1)
-			goto writeit;
-	(void)sprintf(pb, "session=%ld",
+	snprintf(tempstr, sizeof(tempstr), " session=%ld",
 		pjob->ji_wattr[(int)JOB_ATR_session_id].at_val.at_long);
-	i = strlen(pb);
-	pb  += i;
-	len -= i;
+	if (!pbs_strcat(&acct_buf, &acct_bufsize, tempstr))
+		return;
 
 	/* Alternate id if present */
 
 	if (pjob->ji_wattr[(int)JOB_ATR_altid].at_flags & ATR_VFLAG_SET) {
-		i = 9 + strlen(pjob->ji_wattr[(int)JOB_ATR_altid].at_val.at_str);
-		if (i > len)
-			if (grow_acct_buf(&pb, &len, i) == -1)
-				goto writeit;
 #ifdef WIN32
-		(void)sprintf(pb, " alt_id=%s",
-			replace_space(
-			pjob->ji_wattr[(int)JOB_ATR_altid].at_val.at_str,
-			acctlog_spacechar));
+		win_str = replace_space(pjob->ji_wattr[(int)JOB_ATR_altid].at_val.at_str, acctlog_spacechar);
+		if (cpy_quote_value(&acct_buf, &acct_bufsize, win_str, JOB_ALTID_FMT) == -1)
+			return;
 #else
-		(void)sprintf(pb, " alt_id=%s",
-			pjob->ji_wattr[(int)JOB_ATR_altid].at_val.at_str);
+		if (cpy_quote_value(&acct_buf, &acct_bufsize, pjob->ji_wattr[(int)JOB_ATR_altid].at_val.at_str, JOB_ALTID_FMT) == -1)
+			return;
 #endif
-		i = strlen(pb);
-		pb  += i;
-		len -= i;
 	}
 
 	/* add the execution ended time */
-	i = 18;
-	if (i > len)
-		if (grow_acct_buf(&pb, &len, i) == -1)
-			goto writeit;
-	(void)sprintf(pb, " end=%ld", (long)time_now);
-	i = strlen(pb);
-	pb  += i;
-	len -= i;
+	snprintf(tempstr, sizeof(tempstr), " end=%ld", (long)time_now);
```

Comment:
I think you need a space around " end=%ld".

Reply:
No that is fine because the first space is the space between the two attributes in the accounting log, as other printing formats are defined above _FMT are having a space before so for time I am providing a space before while saving.